### DISCUSSION TITLE
[BUG] Error install UTMStack Agent

### DISCUSSION THREAD

**Discussion by Pelican9091** [utmstack/UTMStack#892] [Q&A] [answered]
Same error on multiple systems when installing agents (windows and debian)

2024-10-29T21:07:55.6998089Z ERROR d4bd38e7-4b85-48a2-a1c9-a07cbc7f010a C:/Users/aimng/actions-runner/_work/UTMStack/UTMStack/agent/installer/main.go 70 error downloading dependencies: error getting current version: error getting master version: invalid character '<' looking for beginning of value

**Comment by osmontero** [comment]
This looks like an error response from the UTMStack server, maybe the agent manager was not ready when you tried to install the agent, I strongly recommend to check the logs of the agent manager container if this issue continue. In the mean time lets move this to discussions until a bug is confirmed.

**Comment by c3s4rfred** [comment]
Hi @Pelican9091 this isn't a bug, it is a common network error when installing the agent, check your network connection between the machine where you're trying to install the agent and the UTMStack's server machine.

Best regards

**Reply by BoomaWorld** [reply]
Same error. Ping works fine, port 9000 and 50051 successfully tested from agent side.
But my server dont accessed from Internet, only intranet.

Installing UTMStack Agent...
Downloading UTMStack dependencies...
UTMStack: error: error downloading dependencies: error getting current version: error getting master version: invalid ch
aracter 'T' looking for beginning of value

**Reply by c3s4rfred** [reply]
Hi @BoomaWorld UTMStack needs internet connection to access CDN and download dependencies.

Best regards

**Reply by BoomaWorld** [reply]
Hello! Im understand it ofc, its have connection to Internet, but agents can access it only from intranet. Ports open only into local perimeter. Im still think its bug, or wrong logic. I don,t wanna public my XDR port to download dependencies for agent on end host.

**Reply by BoomaWorld** [reply]
> Hi @BoomaWorld UTMStack needs internet connection to access CDN and download dependencies.
> 
> Best regards

May you answer how i can beat this error?

**Reply by osmontero** [reply]
You are not required to open any port from the internet to your local network device, you need to allow connecting from your local device to the UTMStack CDN; it is a static IP you can add to your firewall rules. (outbound rule, local:any → 34.144.229.76:443)

Further versions will download any agent dependencies from the UTMStack server itself, avoiding this direct connection to our CDN.

**Reply by Kbayero** [reply, marked as answer]
Hello @Pelican9091 @BoomaWorld ,

You can update to version 10.7, which allows you to download the agent's dependencies from the master, so you won't need internet access in your agent machines.

**Comment by SimonDuServ** [comment]
Hi,
My UTMStack agent was working fine until I changed my VM's IP address. After the change, I reinstalled the agent pointing to the new IP, but the dashboard now shows the status as 'Unknown' and logs are no longer flowing

How can I fix this synchronization issue so the agent is recognized again?
